Kernel/Arch/aarch64/Interrupts.cpp

```diff
@@ -5,27 +5,132 @@
  */
 
 #include <Kernel/Arch/Interrupts.h>
+#include <Kernel/Arch/aarch64/InterruptManagement.h>
+#include <Kernel/Interrupts/GenericInterruptHandler.h>
+#include <Kernel/Interrupts/SharedIRQHandler.h>
+#include <Kernel/Interrupts/UnhandledInterruptHandler.h>
+
+struct TrapFrame;
 
 namespace Kernel {
 
-GenericInterruptHandler& get_interrupt_handler(u8)
+static Array<GenericInterruptHandler*, 64> s_interrupt_handlers;
+
+extern "C" void handle_interrupt(TrapFrame const* const);
+extern "C" void handle_interrupt(TrapFrame const* const)
+{
+    for (auto& interrupt_controller : InterruptManagement::the().controllers()) {
+        auto pending_interrupts = interrupt_controller->pending_interrupts();
+
+        // TODO: Add these interrupts as a source of entropy for randomness.
+        u8 irq = 0;
+        while (pending_interrupts) {
+            if ((pending_interrupts & 0b1) != 0b1) {
+                irq += 1;
+                pending_interrupts >>= 1;
+                continue;
+            }
+
+            // TODO: Consider removing passing the RegisterState into the handle_interrupt()
+            //       function, since no IRQHandler seems to be using the registers.
+            RegisterState regs {};
+
+            auto* handler = s_interrupt_handlers[irq];
+            VERIFY(handler);
+            handler->increment_invoking_counter();
+            handler->handle_interrupt(regs);
+            handler->eoi();
+
+            irq += 1;
+            pending_interrupts >>= 1;
+        }
+    }
+}
+
+// FIXME: Share the code below with Arch/x86/common/Interrupts.cpp
```

Comment:
Is introducing a common base class tricky?

But agreed that this is fine for now.

Reply:
I tried it, but it did not turn out nicely. It shouldn't be to tricky, but I just wanted to get this PR out.